src/pytestqt/qt_compat.py

```diff
@@ -153,6 +155,33 @@ def _import_module(module_name):
             self.QStandardItemModel = QtGui.QStandardItemModel
             self.QAbstractListModel = QtCore.QAbstractListModel
             self.QAbstractTableModel = QtCore.QAbstractTableModel
+        else:
+            assert False, "Expected either is_pyqt or is_pyside"
+
+        if self.pytest_qt_api == "pyside2":
+            import shiboken2
+
+            self.isdeleted = lambda obj: not shiboken2.isValid(obj)
+        elif self.pytest_qt_api == "pyside6":
+            import shiboken6
+
+            self.isdeleted = lambda obj: not shiboken6.isValid(obj)
+        else:
+            assert self.is_pyqt
+            self.isdeleted = _import_module("sip").isdeleted
+
+        if self.pytest_qt_api == "pyqt6":
+            # exec was a keyword in Python 2, PyQt6 dropped the .exec_ alias but
```

Comment:
Not sure we should leak out those names to user space... perhaps better to have `self.QCoreApplication_exec = (...` and we use that internally? Same for the other two.

Reply:
Good point, I didn't consider that! Yeah, let's go for a function instead then.